File: src/main/java/com/learn/todolist/config/SwaggerConfig.java
package com.learn.todolist.config;

import io.swagger.v3.oas.annotations.OpenAPIDefinition;
import io.swagger.v3.oas.annotations.enums.SecuritySchemeType;
import io.swagger.v3.oas.annotations.info.Info;
import io.swagger.v3.oas.annotations.security.SecurityRequirement;
import io.swagger.v3.oas.annotations.security.SecurityScheme;
import io.swagger.v3.oas.annotations.security.SecuritySchemes;
import org.springdoc.core.models.GroupedOpenApi;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;


@SecuritySchemes({
        @SecurityScheme(
                name = "bearerAuth",
                type = SecuritySchemeType.HTTP,
                scheme = "bearer",
                bearerFormat = "JWT"
        )
})
@OpenAPIDefinition(
        info = @Info(
                title = "Todo List API",
                version = "1.0",
                description = "Todo List API v1.0"
        ),
        security = @SecurityRequirement(name = "bearerAuth")
)
@Configuration
public class SwaggerConfig {
        private final String paths = "/api/**";

//        @Bean
//        public GroupedOpenApi openApi() {
//                GroupedOpenApi.builder()
//                        .group("todoList")
//                        .pathsToMatch(paths)
//                        .addOpenApiCustomizer(openApi
//        }

}

File: src/main/kotlin/com/todolist/config/swagger/SwaggerConfig.kt
package com.todolist.config.swagger

import io.swagger.v3.oas.models.Components
import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Info

import io.swagger.v3.oas.models.security.SecurityScheme
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class SwaggerConfig {
    private final val SECURITY_SCHEME_NAME = "Authorization"

    @Bean
    fun openApi(): OpenAPI = OpenAPI()
        .components(
            Components()
                .addSecuritySchemes(SECURITY_SCHEME_NAME, securityScheme())
        )
        .info(apiInfo())

    private fun apiInfo(): Info = Info()
        .title("TODO REST APIDocs")
        .description("TODO API")
        .version("1.0.0")

    private fun securityScheme(): SecurityScheme {
        return SecurityScheme()
            .type(SecurityScheme.Type.HTTP)
            .scheme("bearer")
            .bearerFormat("JWT")
    }

}